jerry-debugger/jerry_client.py

```diff
@@ -60,54 +58,49 @@ def do_display(self, args):
 
     def do_break(self, args):
         """ Insert breakpoints on the given lines or functions """
-        result = ""
-        result = self.debugger.set_break(args)
-        if self.debugger.not_empty(result):
-            print(result.get_data())
+        sys.stdout.write(self.debugger.set_break(args))
```

Comment:
what's the reason for rewriting lots of `print`s to `sys.stdout.write`? `print` is a lot more pythonic, `sys` is too low-level. unless there is a good reason, it's generally better practice to use `print`. (even for printing to stderr, if that's needed somewhere.)

Reply:
+1 for keeping the print

Reply:
Same problem as `putstr`, adds an unwanted newline. People recommended to use stdout.write.

Reply:
I'm not completely sure what the unwanted newline may be. This code is printing to the console and users typically expect info to be broken into lines. If this is just part of the output, it may be a signal that this function should not print anything but return a string, perhaps, that can be composed by the caller into a full line of text and printed there.

Anyway, if the only feature that's missing is the no-newline-at-the-end-of-printing, then consult the manual: https://docs.python.org/2/library/functions.html#print . (`print('text to be printed', end='')`.

Reply:
Strings returned by the debugger client contains newlines, including terminating newlines. If you print them with `print` you get two newlines, and that is ugly. The problem with `end=''` that it is python 3. The current one is compatible with any kind of python.

I have a suggestion: I can create a utility, e.g `yield` function:
```
def yield(str):
    sys.stdout.write(str)
```

And use yield everywhere. What do you think?

Reply:
The link I've sent you is Python 2 docs. Did you import print function from future?

As for yield, don't do that. Yield is a keyword in Python 3, used for generators.

Reply:
Yes, print.

```
$ python2
Python 2.7.6 (default, Nov 23 2017, 15:49:48)
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from __future__ import print_function
>>> print('text to be printed', end='')
text to be printed>>>
```

Reply:
I don't like adding `end=''` at the end of each print. I would prefer to introduce a new function.

Reply:
I don't like reinventing the wheel. `print` is there. `end` is there, and it's not Python 3, it is compatible with any kind of Python. Moreover, if you introduce a new function, it'll either be a member of the `DebuggerPrompt` class or a becomes a global function in this module. The first option will mandate a `self.` prefix to all print replacements, which is hardly any better than `end=''`. If it's a global function, then it's ugly and awkward to have a single module-level `print_without_end` utility function that's only used by the class.

I still +1 myself and the `end=''` approach.

Reply:
This function would be in the main program, not in the Debugger module, which should print nothing at all (and no need to know anything about where the output goes). No need for `self` in this utility as well.

Reply:
I don't understand what this means. I see no Debugger python module, and 'main program' (if it means the `main` function) is in this file where we are discussing the the pros and cons of `print` vs utility functions.

If 'main program' stands for the whole jerry_client.py file, then I've already commented on that approach (see: module-level global function used by a single class only -- ugly and awkward).

Reply:
The debugger client has two files at the moment: `jerry_client_ws.py` which is the Debugger Client module, and  `jerry_client.py` application which uses the module to provide the user interface. The user interface is not something reusable, just a simple application. Adding utility functions there is normal.

Reply:
Just because something can be technically done doesn't mean that it is good design. Even the whole class in the jerry_client.py could be dropped and replaced with global functions and some global state. But that also doesn't mean that it is the way to go.

Reply:
The `jerry_client.py` is not a single class file. It contains a class because python command line support designed this way, but that class is just a utility class. It cannot be avoided unless we use another command line system. So this is not a design, it is just forced on us.

The core function is the `main()` function, which is a message loop processor.

Patch updated.